tests/components/google/test_init.py

```diff
@@ -1,67 +1,513 @@
-        yield
 
+@pytest.fixture
+async def code_expiration_delta() -> datetime.timedelta:
+    """Fixture for code expiration time, defaulting to the future."""
+    return datetime.timedelta(minutes=3)
 
-async def test_setup_component(hass, google_setup):
-    """Test setup component."""
-    config = {"google": {CONF_CLIENT_ID: "id", CONF_CLIENT_SECRET: "secret"}}
 
-    assert await async_setup_component(hass, "google", config)
+@pytest.fixture
+async def mock_code_flow(
+    code_expiration_delta: datetime.timedelta,
+) -> YieldFixture[Mock]:
+    """Fixture for initiating OAuth flow."""
+    with patch(
+        "oauth2client.client.OAuth2WebServerFlow.step1_get_device_and_user_codes",
+    ) as mock_flow:
+        mock_flow.return_value.user_code_expiry = utcnow() + code_expiration_delta
+        mock_flow.return_value.interval = CODE_CHECK_INTERVAL
+        yield mock_flow
 
 
-async def test_get_calendar_info(hass, test_calendar):
-    """Test getting the calendar info."""
-    calendar_info = await hass.async_add_executor_job(
-        google.get_calendar_info, hass, test_calendar
+@pytest.fixture
+async def token_scopes() -> list[str]:
+    """Fixture for scopes used during test."""
+    return ["https://www.googleapis.com/auth/calendar"]
+
+
+@pytest.fixture
+async def creds(token_scopes: list[str]) -> OAuth2Credentials:
+    """Fixture that defines creds used in the test."""
+    token_expiry = utcnow() + datetime.timedelta(days=7)
+    return OAuth2Credentials(
+        access_token="ACCESS_TOKEN",
+        client_id="client-id",
+        client_secret="client-secret",
+        refresh_token="REFRESH_TOKEN",
+        token_expiry=token_expiry,
+        token_uri="http://example.com",
+        user_agent="n/a",
+        scopes=token_scopes,
     )
-    assert calendar_info == {
-        "cal_id": "qwertyuiopasdfghjklzxcvbnm@import.calendar.google.com",
-        "entities": [
-            {
-                "device_id": "we_are_we_are_a_test_calendar",
-                "name": "We are, we are, a... Test Calendar",
-                "track": True,
-                "ignore_availability": True,
-            }
-        ],
-    }
-
-
-async def test_found_calendar(hass, google_setup, mock_next_event, test_calendar):
-    """Test when a calendar is found."""
-    config = {
-        "google": {
-            CONF_CLIENT_ID: "id",
-            CONF_CLIENT_SECRET: "secret",
-            "track_new_calendar": True,
+
+
+@pytest.fixture
+async def mock_exchange(creds: OAuth2Credentials) -> YieldFixture[Mock]:
+    """Fixture for mocking out the exchange for credentials."""
+    with patch(
+        "oauth2client.client.OAuth2WebServerFlow.step2_exchange", return_value=creds
+    ) as mock:
+        yield mock
+
+
+@pytest.fixture
+async def token_file(
+    hass: HomeAssistant, creds: OAuth2Credentials, token_filename: Path
+) -> None:
+    """Fixture to populate an existing token file."""
+    storage = Storage(token_filename)
```

Comment:
Isn't it possible to patch out the client so we don't need to worry about writing files at all?

Reply:
Removed. See comments below about additional code changes i had to make to keep this simple.